src/flow-manager.c

```diff
@@ -504,22 +492,36 @@ static uint32_t FlowTimeoutHash(FlowManagerTimeoutThread *td,
     return cnt;
 }
 
+/** \internal
+ *  \brief handle timeout for a slice of hash rows
+ *  If we wrap around we call FlowTimeoutHash twice */
 static uint32_t FlowTimeoutHashInChunks(FlowManagerTimeoutThread *td,
         struct timeval *ts,
         const uint32_t hash_min, const uint32_t hash_max,
-        FlowTimeoutCounters *counters, uint32_t iter, const uint32_t chunks)
+        FlowTimeoutCounters *counters, const uint32_t rows, uint32_t *pos)
 {
-    const uint32_t rows = hash_max - hash_min;
-    const uint32_t chunk_size = rows / chunks;
-
-    const uint32_t min = iter * chunk_size + hash_min;
-    uint32_t max = min + chunk_size;
-    /* we start at beginning of hash at next iteration so let's check
-     * hash till the end */
-    if (iter + 1 == chunks) {
-        max = hash_max;
+    uint32_t start = 0;
+    uint32_t end = 0;
+    uint32_t cnt = 0;
+    uint32_t rows_left = rows;
+
+again:
```

Comment:
Any reason this can't be a while loop?

Reply:
Not really. It's an edge case for the end of the array case.

Reply:
Even with a `do { } while (rows_left)`?  It just looks so much like it could be done with a loop I think it needs an explanation as to why not.